lib/api/CMakeLists.txt

```diff
@@ -104,7 +104,7 @@ if(WASMEDGE_BUILD_SHARED_LIB)
   endif()
 
   target_link_libraries(wasmedge_shared
-    PRIVATE
+    PUBLIC
```

Comment:
Why this should be changed?

Reply:
When using `PRIVATE`, some `undefined symbol` errors are thrown when a dynamic library is being linked with `wasmedge_shared`. (cmake 3.18)

It may be caused by the low version of cmake.

The `wasmedgeCAPI` target is `OBJECT`，it is usually added to `target_sources`, not to `target_link_libraries`.

I suggest the project team to recheck the minimum requirements for the required cmake version.

Reply:
Change this to PUBLIC will make object file included in all tools, which is redundant binary.

Reply:
Use `[[gnu::visibility("default")]]` in implementation to export symbols in library.
See `lib/system/allocator.cpp:60`.

Reply:
cmake 3.21 added something about linking object libraries, which I'm guessing should work.

The Android SDK only supported up to cmake 3.18 in the past, and now it can support cmake 3.22. I'll try again later.